### PR TITLE
Adding custom User-Agent

### PR DESCRIPTION
**Pull request by giorgiosironi**
When investigating for who's calling a certain API, it can be useful. Once agreed on the format, it can be applied to other services.

### CODE REVIEW

File: src/App/Kernel.php [hunk 1]
```diff
@@ -358,8 +359,11 @@ public function dependencies(Application $app)
 
         $app['api.sdk'] = function (Application $app) {
             return new ApiSdk(
-                new Guzzle6HttpClient(
-                    $app['guzzle']
+                new UserAgentPrependingHttpClient(
+                    new Guzzle6HttpClient(
+                        $app['guzzle']
+                    ),
+                    'elife/recommendations'
```
Comment (stephenwf) [inline src/App/Kernel.php]: Is that the sha of the git commit on recommendations, or on the client?

Comment (thewilkybarkid) [inline src/App/Kernel.php]: Recommendations, the client and SDK are already adding their own.

Comment (thewilkybarkid) [inline src/App/Kernel.php]: So a Journal request looks like `eLifeJournal/dev-develop@a35707a eLifeApiSdk/dev-master@2d7a96b eLifeApiClient/dev-master@4aca6b1 GuzzleHttp/6.2.1 curl/7.51.0 PHP/7.1.1`.

Comment (thewilkybarkid) [inline src/App/Kernel.php]: https://github.com/elifesciences/journal/blob/develop/app/config/services.yml#L15-L20

**Comment by giorgiosironi** [comment]
Rebased with the necessary boilerplate to use the version in the header.

**Comment by thewilkybarkid** [comment]
`ocramius/package-versions` should be added as a dependency.

**Comment by giorgiosironi** [comment]
It's inherited transitively by the client, but I'll explicitly specify it.

**Comment by thewilkybarkid** [comment]
Yeah, but your code is now using it. If the client updated to a new major version your code could break.

**Comment by giorgiosironi** [comment]
Added dependency.

**Pull request merged**